cloudferrylib/os/compute/nova_compute.py

```diff
@@ -428,20 +428,56 @@ def _deploy_instances(self, info_compute):
     def create_instance(self, **kwargs):
         return self.nova_client.servers.create(**kwargs).id
 
-    def get_instances_list(self, detailed=True, search_opts=None,
-                           marker=None,
-                           limit=None):
+    def get_instances_list(self, tenant_id=None, network_name=None, **kwargs):
+        """
+        Get list of instances.
+
+        :param tenant_id: Tenant ID
+        :param network_name: Network name
+        Key-value arguments:
+        :param detailed:
+        :param search_opts:
+        :param marker:
+        :param limit:
+
+        :return: Nova Server instance.
+        """
+        search_opts = kwargs.get('search_opts', None)
         ids = search_opts.get('id', None) if search_opts else None
-        if not ids:
-            return self.nova_client.servers.list(detailed=detailed,
-                                                 search_opts=search_opts,
-                                                 marker=marker, limit=limit)
-        else:
+        if ids:
             if type(ids) is list:
                 return [self.nova_client.servers.get(i) for i in ids]
             else:
                 return [self.nova_client.servers.get(ids)]
 
+        if tenant_id:
+            return self.get_instances_list_by_tenant(tenant_id, **kwargs)
+        elif network_name:
+            return self.get_instances_list_by_network(network_name,
+                                                      **kwargs)
+        else:
+            return self.nova_client.servers.list(**kwargs)
+
+    def get_instances_list_by_tenant(self, tenant_id, user_list=None, **kw):
+        instance_list = []
+        search_list = ([self.get_instance(i) for i in user_list]
+                       if user_list
+                       else self.nova_client.servers.list(**kw))
+        for instance in search_list:
+            if instance.tenant_id == tenant_id:
+                instance_list.append(instance)
+        return instance_list
+
+    def get_instances_list_by_network(self, network, user_list=None, **kwargs):
+        instance_list = []
+        search_list = ([self.get_instance(i) for i in user_list]
+                       if user_list
+                       else self.nova_client.servers.list(**kwargs))
+        for instance in search_list:
+            if network in instance.networks:
+                instance_list.append(instance)
+        return instance_list
```

Comment:
`get_instances_list_by_network` and `get_instances_list_by_tenant` look terribly similar, the only difference is the condition (`if <tenant|network> in <instance.tenant|instance.networks>`, please refactor so that there is only one method with condition passed as argument.